instrumentation/net/http/otelhttp/metrics.go

```diff
@@ -0,0 +1,143 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otelhttp
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"sync"
+	"time"
+
+	"go.opentelemetry.io/otel/semconv"
+
+	"go.opentelemetry.io/otel/unit"
+
+	"go.opentelemetry.io/otel/api/metric"
+	"go.opentelemetry.io/otel/label"
+)
+
+type metricsTransport struct {
+	meter                  metric.Meter
+	base                   *Transport
+	clientDurationRecorder metric.Float64ValueRecorder
+}
+
+type tracker struct {
+	ctx     context.Context
+	start   time.Time
+	body    io.ReadCloser
+	endOnce sync.Once
+	labels  []label.KeyValue
+
+	// recorders
+	clientDurationRecorder metric.Float64ValueRecorder
+}
+
+func (trans *metricsTransport) applyConfig(c *config) {
+	trans.base.applyConfig(c)
+
+	trans.meter = c.Meter
+	trans.createMeasures()
+}
+
+// RoundTrip implements http.RoundTripper, delegating to Base and recording stats for the request.
+func (trans *metricsTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	labels := semconv.HTTPClientAttributesFromHTTPRequest(req)
+
+	ctx := req.Context()
+	tracker := &tracker{
+		start:                  time.Now(),
+		ctx:                    ctx,
+		clientDurationRecorder: trans.clientDurationRecorder,
+	}
+
+	// Perform request.;
+	resp, err := trans.base.RoundTrip(req)
+	if err != nil {
+		tracker.labels = append(labels, semconv.HTTPAttributesFromHTTPStatusCode(http.StatusInternalServerError)...)
```

Comment:
This should include the response status not `http.StatusInternalServerError`

Reply:
In case of HTTP connection error occurred, should we recorded nothing or 500 as the HTTP status code. What's your opinion?